hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoCreateFeeCalculator.java

```diff
@@ -0,0 +1,76 @@
+
+import static java.util.Objects.requireNonNull;
+import static org.hiero.hapi.fees.FeeScheduleUtils.lookupServiceFee;
+
+import com.hedera.hapi.node.base.HederaFunctionality;
+import com.hedera.hapi.node.base.Key;
+import com.hedera.hapi.node.base.SubType;
+import com.hedera.hapi.node.transaction.Query;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.spi.fees.AbstractSimpleFeeCalculator;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.hiero.hapi.fees.FeeResult;
+
+/** Calculates CryptoCreate fees. Per HIP-1261, uses SIGNATURES and KEYS extras. */
+public class CryptoCreateFeeCalculator extends AbstractSimpleFeeCalculator {
+
+    @Override
+    @NonNull
+    public FeeResult calculateTxFee(@NonNull final TransactionBody body, @Nullable final TxContext context) {
+        requireNonNull(body, "body");
+        final var op = body.cryptoCreateAccountOrThrow();
+        // Count keys in the account being created
+        final long keyCount = op.hasKey() ? countKeys(op.key()) : 0L;
+
+        // Build usage mapper - fluent API makes fee structure explicit
+        final var usageMapper = usageBuilder()
+                .withSignatures(context.numTxnSignatures())
+                .withKeys(keyCount)
+                .build();
+
+        // Get fee schedule
+        final var feeCalculator = context.feeCalculatorFactory().feeCalculator(SubType.DEFAULT);
+        final var feeSchedule = feeCalculator.getSimpleFeesSchedule();
+        final var serviceDef = lookupServiceFee(feeSchedule, HederaFunctionality.CRYPTO_CREATE);
+
+        // Calculate fees
+        long nodeFee = feeSchedule.node().baseFee()
+                + calculateExtraFees(feeSchedule.node().extras(), feeSchedule, usageMapper);
+
+        long serviceFee = serviceDef.baseFee() + calculateExtraFees(serviceDef.extras(), feeSchedule, usageMapper);
+
+        long networkFee = calculateNetworkFee(nodeFee, feeSchedule.network().multiplier());
+
+        // Return result
+        final var result = new FeeResult();
+        result.node = nodeFee;
+        result.network = networkFee;
+        result.service = serviceFee;
+        return result;
+    }
+
+    /**
+     * Counts all keys including nested ones in threshold/key lists.
+     */
+    private long countKeys(Key key) {
+        return switch (key.key().kind()) {
+            case ED25519, ECDSA_SECP256K1, ECDSA_384 -> 1L;
+            case THRESHOLD_KEY ->
+                key.thresholdKeyOrThrow().keys().keys().stream()
+                        .mapToLong(this::countKeys)
+                        .sum();
+            case KEY_LIST ->
+                key.keyListOrThrow().keys().stream().mapToLong(this::countKeys).sum();
+            default -> 0L;
+        };
+    }
+
+    @Override
+    @NonNull
+    public FeeResult calculateQueryFee(@NonNull final Query query, @Nullable final QueryContext context) {
```

Comment:
Why do we need to override here ? By default this throws UnsupportedOperationException right?